tests/test_data/test_transform.py

```diff
@@ -688,3 +688,13 @@ def test_mosaic():
     mosaic_module = build_from_cfg(transform, PIPELINES)
     results = mosaic_module(results)
     assert results['img'].shape[:2] == (20, 24)
+
+
+def test_blur():
+    # TODO: Figure out how to run tests
```

Comment:
Hi, I installed the dependencies listed there and tried to run the tests as-is, without adding any new tests. I got this error about `mmcv`. I have `mmcv` installed as well, but is there maybe a particular version I should be testing with?

```
eric@desktop:~/Desktop/mmsegmentation$ pytest tests/test_data/test_transform.py
=========================================================================================== test session starts ===========================================================================================
platform linux -- Python 3.8.10, pytest-7.1.2, pluggy-1.0.0
rootdir: /home/eric/Desktop/mmsegmentation, configfile: pytest.ini
plugins: xdoctest-1.0.1
collected 0 items / 1 error                                                                                                                                                                               

================================================================================================= ERRORS ==================================================================================================
___________________________________________________________________________ ERROR collecting tests/test_data/test_transform.py ____________________________________________________________________________
ImportError while importing test module '/home/eric/Desktop/mmsegmentation/tests/test_data/test_transform.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/usr/lib/python3.8/importlib/__init__.py:127: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_data/test_transform.py:8: in <module>
    from mmcv.utils import build_from_cfg
E   ImportError: cannot import name 'build_from_cfg' from 'mmcv.utils' (/home/eric/.local/lib/python3.8/site-packages/mmcv/utils/__init__.py)
========================================================================================= short test summary info =========================================================================================
ERROR tests/test_data/test_transform.py
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================================================ 1 error in 0.21s =============================================================================================
```

Reply:
Hi, I test it with `mmcv-full==1.6.1`. Could you import `mmcv` at any other python files?
For example:
```python
from mmcv.utils import build_from_cfg

if __name__ == '__main__':
    print('hello')

```

Reply:
Hm, this is getting tricky. Does this ring any bells? The `mmcv` version that you mentioned doesn't seem to have `build_from_cfg` when I install it.

```
eric@desktop:~/Desktop/mmsegmentation$ pip install mmcv-full==1.6.1
Collecting mmcv-full==1.6.1
  Downloading mmcv-full-1.6.1.tar.gz (563 kB)
     |████████████████████████████████| 563 kB 5.0 MB/s 
Requirement already satisfied: Pillow in /home/eric/.local/lib/python3.8/site-packages (from mmcv-full==1.6.1) (9.2.0)
Requirement already satisfied: addict in /home/eric/.local/lib/python3.8/site-packages (from mmcv-full==1.6.1) (2.4.0)
Requirement already satisfied: numpy in /home/eric/.local/lib/python3.8/site-packages (from mmcv-full==1.6.1) (1.23.1)
Requirement already satisfied: packaging in /home/eric/.local/lib/python3.8/site-packages (from mmcv-full==1.6.1) (21.3)
Requirement already satisfied: pyyaml in /home/eric/.local/lib/python3.8/site-packages (from mmcv-full==1.6.1) (6.0)
Collecting yapf
  Downloading yapf-0.32.0-py2.py3-none-any.whl (190 kB)
     |████████████████████████████████| 190 kB 40.1 MB/s 
Requirement already satisfied: pyparsing!=3.0.5,>=2.0.2 in /home/eric/.local/lib/python3.8/site-packages (from packaging->mmcv-full==1.6.1) (3.0.9)
Building wheels for collected packages: mmcv-full
  Building wheel for mmcv-full (setup.py) ... done
  Created wheel for mmcv-full: filename=mmcv_full-1.6.1-py2.py3-none-any.whl size=860342 sha256=1ec28279b3ea0ea4b7f00901fe930afb4faa5b7fd61f43c92748ae2288907955
  Stored in directory: /home/eric/.cache/pip/wheels/05/48/31/8aacd3b86a76e602057c5c1dd234c8bbf7efcce5bec821ea4c
Successfully built mmcv-full
Installing collected packages: yapf, mmcv-full
Successfully installed mmcv-full-1.6.1 yapf-0.32.0
eric@desktop:~/Desktop/mmsegmentation$ ipython
iPython 3.8.10 (default, Jun 22 2022, 20:18:18) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.4.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import mmcv

In [2]: from mmcv.utils import build_from_cfg
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
Input In [2], in <cell line: 1>()
----> 1 from mmcv.utils import build_from_cfg

ImportError: cannot import name 'build_from_cfg' from 'mmcv.utils' (/home/eric/.local/lib/python3.8/site-packages/mmcv/utils/__init__.py)

In [3]: mmcv.__version__
Out[3]: '1.6.1'
```

Reply:
It's really weird.
```shell
$ ipython
Python 3.8.10 (default, May 19 2021, 11:01:55) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.26.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import mmcv

In [2]: mmcv.__version__
Out[2]: '1.6.1'

In [3]: from mmcv.utils import build_from_cfg

In [4]: 
```
Could you try reinstalling with `--no-cache` argument?
```shell
pip install mmcv-full==1.6.1 --no-cache
```

In all `MM` frameworks, most of the components are built by the `build_from_cfg` API.

Reply:
I'm sorry, it's still not working and I'm not sure why. I tried installing as you suggested and still have the issue. The only thing I can think to do is open an mmcv issue, which is here: https://github.com/open-mmlab/mmcv/issues/2213. Hopefully they can help resolve this.

```
eric@laptop:~$ pip install mmcv-full==1.6.1 --no-cache-dir
/usr/lib/python3/dist-packages/secretstorage/dhcrypto.py:15: CryptographyDeprecationWarning: int_from_bytes is deprecated, use int.from_bytes instead
  from cryptography.utils import int_from_bytes
/usr/lib/python3/dist-packages/secretstorage/util.py:19: CryptographyDeprecationWarning: int_from_bytes is deprecated, use int.from_bytes instead
  from cryptography.utils import int_from_bytes
Defaulting to user installation because normal site-packages is not writeable
Collecting mmcv-full==1.6.1
  Downloading mmcv-full-1.6.1.tar.gz (563 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 563.2/563.2 kB 11.9 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
Requirement already satisfied: addict in ./.local/lib/python3.8/site-packages (from mmcv-full==1.6.1) (2.4.0)
Requirement already satisfied: numpy in ./.local/lib/python3.8/site-packages (from mmcv-full==1.6.1) (1.22.2)
Requirement already satisfied: packaging in /usr/lib/python3/dist-packages (from mmcv-full==1.6.1) (20.3)
Requirement already satisfied: Pillow in ./.local/lib/python3.8/site-packages (from mmcv-full==1.6.1) (8.3.1)
Requirement already satisfied: pyyaml in ./.local/lib/python3.8/site-packages (from mmcv-full==1.6.1) (6.0)
Requirement already satisfied: yapf in ./.local/lib/python3.8/site-packages (from mmcv-full==1.6.1) (0.32.0)
Building wheels for collected packages: mmcv-full
  Building wheel for mmcv-full (setup.py) ... done
  Created wheel for mmcv-full: filename=mmcv_full-1.6.1-py2.py3-none-any.whl size=860366 sha256=3ef6e4dc050eb780f39a8ed6a134a88053ecedda1d5ca616919790b0b804dd3f
  Stored in directory: /tmp/pip-ephem-wheel-cache-6j7k8uqk/wheels/05/48/31/8aacd3b86a76e602057c5c1dd234c8bbf7efcce5bec821ea4c
Successfully built mmcv-full
Installing collected packages: mmcv-full
Successfully installed mmcv-full-1.6.1

eric@laptop:~$ ipy
Python 3.8.10 (default, Jun 22 2022, 20:18:18) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.13.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import mmcv; print(mmcv.__version__); from mmcv.utils import build_from_cfg

1.6.1

---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-1-8865daaf63bb> in <module>
----> 1 import mmcv; print(mmcv.__version__); from mmcv.utils import build_from_cfg

ImportError: cannot import name 'build_from_cfg' from 'mmcv.utils' (/home/eric/.local/lib/python3.8/site-packages/mmcv/utils/__init__.py)
```

Reply:
Figured out issue with the help of mmcv repo, the correct way to install `mmcv` to get this import was `pip install openmim && mim install mmcv-full`

Reply:
Hello @xiexinch sorry for the slow response. I have one more question about testing. After install mmcv correctly, as well as the requirements you mentioned, the `pytest` command seems to have some issues. Below is what happens when I run it with the default tests, before I have added anything.

What version of pytest are you using? Maybe the version I have is too new and has xdoctest as deprecated?

```
eric@laptop:~/Desktop/mmsegmentation$ pytest tests/test_data/test_transform.py
=========================================================================================== test session starts ===========================================================================================
platform linux -- Python 3.8.10, pytest-7.1.2, pluggy-0.13.0
rootdir: /home/eric/Desktop/mmsegmentation, configfile: pytest.ini
plugins: xdoctest-0.11.0, anyio-3.3.1, colcon-core-0.10.0, cov-2.8.1
collected 0 items / 1 error                                                                                                                                                                               

================================================================================================= ERRORS ==================================================================================================
______________________________________________________________________________________ ERROR collecting test session ______________________________________________________________________________________
Direct construction of xdoctest.plugin.XDoctestModule has been deprecated, please use xdoctest.plugin.XDoctestModule.from_parent.
See https://docs.pytest.org/en/stable/deprecations.html#node-construction-changed-to-node-from-parent for more details.
========================================================================================= short test summary info =========================================================================================
ERROR 
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
============================================================================================ 1 error in 0.11s =============================================================================================

```